### ISSUE TITLE
Deck generation

### ISSUE THREAD

**Issue by CrusaderSVK287** [CrusaderSVK287/Spider-Solitaire#1]
Improve the generation of cards in stock in order to make generated games more likely to be winnable, particular cases would be:
- not having 3 or more of the same card beside each other
- not having 4 or more same cards in the same row (starting from the first row, fifth card)
- not having 6 or more of the same values of cards in the same row (starting from the first row, fifth card)
- Kings not being present at the end of the deck since they can easily bottleneck the entire game, especially when combined with the previous point

In summary, implementing a set of rules to make generated decks more winnable
